### ISSUE TITLE
RuntimeError during fine-tuning conservative model with donated buffer issue

### ISSUE THREAD

**Issue by Knitua** [orbital-materials/orb-models#111]
Hi Orb team,

I’m currently trying to fine-tune the orb_v3_conservative_inf_omat model using your provided finetune.py script. However, I encountered a RuntimeError related to donated buffers during the backward pass:"This backward function was compiled with non-empty donated buffers which requires create_graph=False and retain_graph=False. Please keep backward(create_graph=False, retain_graph=False) across all backward() function calls, or set torch._functorch.config.donated_buffer=False to disable donated buffer."

How can I solve this?
I’ve tried that the loss.backward(create_graph=False, retain_graph=False) call is being used, and I’ve also added os.environ["TORCH_DISABLE_DONATED_BUFFERS"] = "1" at the top of the script. Despite this, the error still occurs. What’s interesting is that the error only happens when using the conservative version of the model (orb_v3_conservative_inf_omat). When switching to the direct version (e.g., orb_v3_direct_inf_omat), the same script runs without issue.

My environment is as follows:

PyTorch version: 2.7.1+cu126

CUDA: 12.6

GPU: NVIDIA H800

OS: Ubuntu 22.04

The command I used to reproduce this error is:

python finetune.py \
  --dataset finetune_train \
  --data_path ./finetune.db \
  --base_model orb_v3_conservative_inf_omat

Given this context, I would like to ask:

How can I solve this problems?
Is this a known compatibility issue with newer versions of PyTorch (or functorch)?
Is there a recommended PyTorch version or workaround to avoid this issue?

Thanks a lot for your great work on orb-models—the architecture and pretraining are impressive and very useful for the community. I’d greatly appreciate any guidance you can offer!

**Comment by ameya98** [comment]
@vsimkus you might be interested in our JAMUN codebase, specifically https://github.com/prescient-design/jamun/blob/main/src/jamun/model/energy.py, where we train an energy model with double backward working with torch.compile!
